### DISCUSSION TITLE
Add delay on AMQP 0-9-1 `basic.get_empty` response

### DISCUSSION THREAD

**Discussion by LoisSotoLopez** [rabbitmq/rabbitmq-server#12255] [Other]
### Is your feature request related to a problem? Please describe.

We would like to have a mechanism for delying the response of `basic.get_empty` messages.

Developers o client applications may implement polling loops that just consume from a queue using `basic.get` AMQP message, process the response and execute the next iteration.

When the consumed queue is empty, those loop iterations may execute too quickly, issuing an excessive number of requests and increasing the CPU usage on the server.

This is of course a bad practice, but still happens.

### Describe the solution you'd like

Similarly to what is done [to prevent DOS due to repeated failed logins from a client](https://github.com/rabbitmq/rabbitmq-server/blob/e396dd130b02967fbb8bb8925d9ecbed845b3134/deps/rabbit/src/rabbit_reader.erl#L860) (PRs [here](https://github.com/rabbitmq/rabbitmq-server/pull/11831) and [here](https://github.com/rabbitmq/rabbitmq-server/pull/11900/files)) we wanted to suggest some similar mechanism to mitigate the effect of `basic.get` client tight loops.

A configurable delay, defaulting to 0, could be added before responding with a `basic.get_empty` message.

### Additional context

If limit-based events (https://github.com/rabbitmq/rabbitmq-server/issues/11207) get implemented we could use them to detect situations subject of causing a DOS, like the one described above.

**Comment by michaelklishin** [comment]
So we need to bastardize RabbitMQ one more time in order to avoid catastrophically bad application-level decisions we are not responsible for.

How about a mechanism to disable `basic.get` entirely instead? This would be the right thing to do if you ask me.
